src/main/kotlin/nl/info/zac/admin/ZaaktypeBpmnConfigurationBeheerService.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-FileCopyrightText: 2025 INFO.nl
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package nl.info.zac.admin
+
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.transaction.Transactional
+import jakarta.transaction.Transactional.TxType.REQUIRES_NEW
+import nl.info.zac.admin.exception.ZaaktypeConfigurationNotFoundException
+import nl.info.zac.admin.model.ZaaktypeBpmnConfiguration
+import nl.info.zac.admin.model.ZaaktypeBpmnConfiguration.Companion.CREATIEDATUM_VARIABLE_NAME
+import nl.info.zac.admin.model.ZaaktypeBpmnConfiguration.Companion.PRODUCTAANVRAAGTYPE_VARIABLE_NAME
+import nl.info.zac.admin.model.ZaaktypeBpmnConfiguration.Companion.ZAAKTYPE_OMSCHRIJVING
+import nl.info.zac.admin.model.ZaaktypeBpmnConfiguration.Companion.ZAAKTYPE_UUID_VARIABLE_NAME
+import nl.info.zac.util.AllOpen
+import nl.info.zac.util.NoArgConstructor
+import java.util.UUID
+import java.util.logging.Logger
+import kotlin.jvm.optionals.getOrNull
+
+@ApplicationScoped
+@Transactional
+@NoArgConstructor
+@AllOpen
+class ZaaktypeBpmnConfigurationBeheerService @Inject constructor(
+    private val entityManager: EntityManager
+) {
+    companion object {
+        private val LOG = Logger.getLogger(ZaaktypeBpmnConfigurationBeheerService::class.java.name)
+    }
+
+    fun storeConfiguration(zaaktypeBpmnConfiguration: ZaaktypeBpmnConfiguration): ZaaktypeBpmnConfiguration {
+        zaaktypeBpmnConfiguration.id?.let {
+            if (findConfiguration(zaaktypeBpmnConfiguration.zaaktypeUuid) == null) {
+                LOG.warning("BPMN configuration with zaaktype UUID '$it' not found, creating new configuration")
+                zaaktypeBpmnConfiguration.id = null
+            }
+        }
+
+        return if (zaaktypeBpmnConfiguration.id != null) {
+            entityManager.merge(zaaktypeBpmnConfiguration)
```

Comment:
I forgot a bit how JPA works but shouldn't you also do a flush here? Or: why do you need to do a flush in the other case but not on a merge?

Reply:
I would say that a store function should only store the thing, not retrieve it? If the clients really needs to retrieve the newly stored configuration they should call findConfiguration themselves maybe? Bit debatable this I guess.